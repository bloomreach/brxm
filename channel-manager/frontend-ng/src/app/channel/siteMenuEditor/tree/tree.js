/*
 * Copyright 2016-2017 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import uiTree from 'angular-ui-tree';
import hippoTreeDirective from './tree.directive';
import itemsRendererDirective from './itemsRenderer/itemsRenderer.directive';
import hippoTreeTemplateDirective from './tree.template.directive';
import HippoTreeCtrl from './tree.controller';

const uiTreeModule = angular
  .module('hippo-cm.ui.tree', [
    uiTree,
  ])
  .directive('hippoTree', hippoTreeDirective)
  .directive('itemsRenderer', itemsRendererDirective)
  .directive('hippoTreeTemplate', hippoTreeTemplateDirective)
  .controller('HippoTreeCtrl', HippoTreeCtrl);

export default uiTreeModule;
