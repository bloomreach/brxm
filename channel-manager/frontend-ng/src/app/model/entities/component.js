/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as HstConstants from '../constants';
import { ComponentEntity } from './component-entity';

export class Component extends ComponentEntity {
  hasExperiment() {
    return !!this.getExperimentId();
  }

  getExperimentId() {
    return this._meta[HstConstants.EXPERIMENT_ID];
  }

  getExperimentStateLabel() {
    if (!this.hasExperiment()) {
      return null;
    }

    return `EXPERIMENT_LABEL_${this._meta[HstConstants.EXPERIMENT_STATE]}`;
  }

  getContainer() {
    return this.container;
  }

  setContainer(container) {
    this.container = container;
  }

  getType() {
    return 'component';
  }

  getRenderVariant() {
    return this._meta[HstConstants.RENDER_VARIANT] || HstConstants.DEFAULT_RENDER_VARIANT;
  }

  getReferenceNamespace() {
    return this._meta[HstConstants.REFERENCE_NAMESPACE];
  }
}
