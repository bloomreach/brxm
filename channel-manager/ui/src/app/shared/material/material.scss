/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~@angular/material/theming';
@import '~@bloomreach/frontend-theme/palettes';
@import 'variables';

$custom-typography: mat-typography-config(
  $font-family: inspect($font-family)
);

@include mat-core($custom-typography);

$primary: mat-palette($blue);
$accent: mat-palette($orange);
$warn: mat-palette($red);
$theme: mat-light-theme($primary, $accent, $warn);

@include angular-material-theme($theme);

// default body text color
.mat-typography {
  color: map-get($text-color, "dark");
}
