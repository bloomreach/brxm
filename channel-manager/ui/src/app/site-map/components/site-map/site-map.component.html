<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->

<mat-form-field floatLabel="never"
                class="mat-elevation-z2 searchbar"
                appearance="none">
  <input matInput
         class="qa-sitemap-search-input"
         [placeholder]="'PLACEHOLDER_SEARCH' | translate "
         [ngModel]="search$ | async"
         (ngModelChange)="search$.next($event)" />
  <button mat-icon-button
          matPrefix
          disabled>
    <mat-icon matPrefix
              fontIcon="mdi-magnify"></mat-icon>
  </button>
  <button *ngIf="search$ | async"
          mat-icon-button
          matSuffix>
    <mat-icon fontIcon="mdi-close"
              (click)="search$.next('')"></mat-icon>
  </button>
</mat-form-field>

<mat-tree #tree
          [dataSource]="dataSource"
          [treeControl]="treeControl">
  <mat-tree-node *matTreeNodeDef="let node"
                 matTreeNodePadding
                 matTreeNodePaddingIndent="24"
                 (click)="selectNode(node)"
                 class="mat-tree-node {{ 'qa-tree-node-name-' + node.name }}"
                 [class.selected]="isSelected(node)"
                 [class.hidden]="!isMatched(node) && !isVisible(node)"
                 [class.disabled]="!isMatched(node)">
    <!-- Trick to get extra padding equal to other nodes with icons -->
    <mat-icon *ngIf="!node.expandable"></mat-icon>

    <mat-icon svgIcon="chevron-right"
              *ngIf="node.expandable"
              matTreeNodeToggle
              [class.expanded]="treeControl.isExpanded(node)">
    </mat-icon>

    {{ node.name }}
  </mat-tree-node>
</mat-tree>
