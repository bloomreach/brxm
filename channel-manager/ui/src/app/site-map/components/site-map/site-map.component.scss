/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import 'variables';
@import 'mixins';
@import 'material-theme';

.mat-icon {
  transition: transform $transition-small-area-speed;
}

.mat-icon.expanded {
  transform: rotate(90deg);
}

.mat-tree-node {
  &.disabled {
    color: $dark-disabled-text;
  }

  &.hidden {
    display: none;
  }

  &.selected {
    background: map-get($blue, 50);
  }

  &:hover {
    cursor: pointer;
  }

  &:not(.selected):hover {
    @include mat-hover-background($theme);
  }
}

.searchbar {
  width: 100%;

  .mat-icon-button {
    color: mat-color($grey, 600);
  }

  ::ng-deep {
    .mat-form-field-wrapper {
      padding: .5em 1em;
    }

    .mat-form-field-flex {
      align-items: center;
    }

    .mat-form-field-infix {
      border-top: none;
      padding: 0;
    }

    .mat-input-element {
      height: 20px;
    }
  }
}
