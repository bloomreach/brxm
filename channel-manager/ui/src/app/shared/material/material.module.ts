/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule, MatIconRegistry } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { DomSanitizer } from '@angular/platform-browser';

import { registerSvgIcons } from './register-svg-icons';

@NgModule({
  exports: [
    MatButtonModule,
    MatListModule,
    MatIconModule,
  ],
})
export class MaterialModule {
  constructor(private readonly iconRegistry: MatIconRegistry, private readonly domSanitizer: DomSanitizer) {
    this.registerCustomIcons();
  }

  private registerCustomIcons(): void {
    registerSvgIcons(this.iconRegistry, this.domSanitizer);
  }
}
