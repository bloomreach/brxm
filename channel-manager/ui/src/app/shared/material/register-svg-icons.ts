/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// tslint:disable:match-default-export-name
import alertCircleOutline from '!!raw-loader!./icons/alert-circle-outline.svg';
import alertOutline from '!!raw-loader!./icons/alert-outline.svg';
import calendarClock from '!!raw-loader!./icons/calendar-clock.svg';
import chevronRight from '!!raw-loader!./icons/chevron-right.svg';
import commentCheckOutline from '!!raw-loader!./icons/comment-check-outline.svg';
import commentProcessingOutline from '!!raw-loader!./icons/comment-processing-outline.svg';
import commentRemoveOutline from '!!raw-loader!./icons/comment-remove-outline.svg';
import lockOutline from '!!raw-loader!./icons/lock-outline.svg';
import minusCircleOutline from '!!raw-loader!./icons/minus-circle-outline.svg';
import plus from '!!raw-loader!./icons/plus.svg';
import xpageIcon from '!!raw-loader!./icons/xpage.svg';
import { MatIconRegistry } from '@angular/material/icon';
import { DomSanitizer } from '@angular/platform-browser';

export function registerSvgIcons(iconRegistry: MatIconRegistry, donSanitizer: DomSanitizer): void {
  const registerIcon = (name: string, svg: string) => {
    iconRegistry.addSvgIconLiteral(name, donSanitizer.bypassSecurityTrustHtml(svg));
  };

  registerIcon('xpage', xpageIcon);
  registerIcon('minus-circle-outline', minusCircleOutline);
  registerIcon('alert-outline', alertOutline);
  registerIcon('comment-processing-outline', commentProcessingOutline);
  registerIcon('comment-remove-outline', commentRemoveOutline);
  registerIcon('calendar-clock', calendarClock);
  registerIcon('alert-circle-outline', alertCircleOutline);
  registerIcon('lock-outline', lockOutline);
  registerIcon('comment-check-outline', commentCheckOutline);
  registerIcon('plus', plus);
  registerIcon('chevron-right', chevronRight);
}
