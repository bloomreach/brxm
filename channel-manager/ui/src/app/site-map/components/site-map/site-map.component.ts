/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FlatTreeControl } from '@angular/cdk/tree';
import { AfterViewChecked, Component, ElementRef, Input, NgZone, OnChanges, OnDestroy, SimpleChanges } from '@angular/core';
import { MatTreeFlatDataSource, MatTreeFlattener } from '@angular/material/tree';
import { Subject } from 'rxjs';
import { debounceTime, distinctUntilChanged, map, pluck } from 'rxjs/operators';

import { IframeService } from '../../../channels/services/iframe.service';
import { SiteMapItem } from '../../models/site-map-item.model';

interface SiteMapItemNode extends SiteMapItem {
  expandable: boolean;
  level: number;
}

@Component({
  selector: 'em-site-map',
  templateUrl: './site-map.component.html',
  styleUrls: ['./site-map.component.scss'],
})
export class SiteMapComponent implements OnChanges, OnDestroy, AfterViewChecked {
  @Input() siteMap: SiteMapItem[] = [];
  @Input() renderPathInfo?: string;

  private readonly treeFlattener = new MatTreeFlattener(
    (node: SiteMapItem, level: number) => ({
      ...node,
      level,
      expandable: !!node.children && node.children.length > 0,
    }),
    node => node.level,
    node => node.expandable,
    node => node.children,
  );
  readonly treeControl = new FlatTreeControl<SiteMapItemNode>(node => node.level, node => node.expandable);
  readonly dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener);

  readonly search$ = new Subject<string>();

  private readonly matched = new Set<SiteMapItemNode>();
  private readonly visible = new Set<SiteMapItemNode>();

  isMatched = this.matched.has.bind(this.matched);
  isVisible = this.visible.has.bind(this.visible);

  constructor(
    private readonly iframeService: IframeService,
    private readonly zone: NgZone,
    private readonly elementRef: ElementRef,
  ) {
    this.search$
      .pipe(
        debounceTime(500),
        map<string, [string, SiteMapItemNode[]]>(query => [query, this.treeControl.dataNodes]),
        distinctUntilChanged(([prevQuery, prevNodes], [query, nodes]) => query === prevQuery && nodes === prevNodes),
        pluck(0),
      )
      .subscribe(this.onSearch.bind(this));
  }

  ngOnChanges(changes: SimpleChanges): void {
    if (changes.siteMap) {
      this.dataSource.data = this.siteMap;
      this.zone.run(() => this.search$.next(''));
    }

    this.expandSelected();
  }

  ngAfterViewChecked(): void {
    const selectedNode = this.elementRef.nativeElement.querySelector('.selected');
    selectedNode?.scrollIntoView();
  }

  ngOnDestroy(): void {
    this.search$.complete();
  }

  isSelected(node: SiteMapItem): boolean {
    return node.renderPathInfo === this.renderPathInfo;
  }

  async selectNode(node: SiteMapItemNode): Promise<void> {
    if (node.expandable) {
      this.treeControl.expand(node);
    }

    if (node.renderPathInfo) {
      await this.iframeService.load(node.renderPathInfo);
    }
  }

  private expandSelected(): void {
    const node = this.treeControl.dataNodes.find(({ renderPathInfo }) => renderPathInfo === this.renderPathInfo);

    if (node) {
      this.treeControl.expand(node);
      this.getParents(node).forEach(parent => {
        this.treeControl.expand(parent);
      });
    }
  }

  private getParents(child: SiteMapItemNode): SiteMapItemNode[] {
    const parents = [];

    for (let i = this.treeControl.dataNodes.indexOf(child), childLevel = this.treeControl.getLevel(child); i >= 0; i--) {
      const node = this.treeControl.dataNodes[i];
      const level = this.treeControl.getLevel(node);
      if (level < childLevel) {
        parents.push(node);
        childLevel = level;
      }
    }

    return parents;
  }

  private onSearch(value: string): void {
    this.matched.clear();
    this.visible.clear();

    if (!value) {
      this.treeControl.dataNodes.forEach(node => {
        this.matched.add(node);
        this.visible.add(node);
      });

      return;
    }

    this.treeControl.dataNodes
      .filter(
        ({ name, pageTitle }) => name.toLowerCase().includes(value.toLowerCase())
          || pageTitle?.toLowerCase().includes(value.toLowerCase()),
      )
      .forEach(node => {
        this.matched.add(node);

        this.getParents(node).forEach(parent => {
          this.treeControl.expand(parent);
          this.visible.add(parent);
        });

        this.treeControl.getDescendants(node).forEach(child => {
          this.visible.add(child);
        });
      });
  }
}
