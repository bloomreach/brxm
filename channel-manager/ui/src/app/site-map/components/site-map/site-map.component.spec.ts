/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { CUSTOM_ELEMENTS_SCHEMA, SimpleChange } from '@angular/core';
import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { MatIconModule } from '@angular/material/icon';
import { MatIconTestingModule } from '@angular/material/icon/testing';
import { MatTreeModule } from '@angular/material/tree';
import { TranslateModule } from '@ngx-translate/core';

import { IframeService } from '../../../channels/services/iframe.service';
import { SiteMapItemMock } from '../../models/site-map-item.model.mock';

import { SiteMapComponent } from './site-map.component';

describe('SiteMapComponent', () => {
  let component: SiteMapComponent;
  let componentEl: HTMLElement;
  let fixture: ComponentFixture<SiteMapComponent>;
  let iframeService: IframeService;

  const mockRenderPathInfo = '/idTest2/idTest3';

  const mockSiteMapTree = [
    new SiteMapItemMock({
      children: [
        new SiteMapItemMock({
          id : 'idTest2',
          name : 'nameTest2',
          pageTitle : null,
          pathInfo : 'idTest2',
          renderPathInfo : '/idTest2',
          children: [
            new SiteMapItemMock({
              id : 'idTest3',
              name : 'nameTest3',
              pageTitle : 'pageTitleTest3',
              pathInfo : 'idTest2/idTest3',
              renderPathInfo : mockRenderPathInfo,
            }),
          ],
        }),
        new SiteMapItemMock({
          id : 'idTest4',
          name : 'nameTest4',
          pageTitle : 'pageTitleTest',
          pathInfo : 'idTest4',
          renderPathInfo : '/idTest4',
        }),
      ],
    }),
  ];

  beforeEach(async(() => {
    const iframeServiceMock = {
      load: jest.fn(() => Promise.resolve()),
    };

    TestBed.configureTestingModule({
      declarations: [
        SiteMapComponent,
      ],
      imports: [
        FormsModule,
        MatTreeModule,
        MatIconModule,
        MatIconTestingModule,
        TranslateModule.forRoot(),
      ],
      providers: [
        { provide: IframeService, useValue: iframeServiceMock },
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();

    iframeService = TestBed.inject(IframeService);
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SiteMapComponent);
    component = fixture.componentInstance;
    componentEl = fixture.nativeElement;
  });

  describe('no tree items', () => {
    it('should display empty tree', () => {
      component.siteMap = [];
      component.ngOnChanges({
        siteMap: {  } as SimpleChange,
      });

      fixture.detectChanges();

      expect(componentEl.querySelector('mat-tree')).toMatchSnapshot();
    });
  });

  describe('tree behavior', () => {
    beforeEach(() => {
      component.siteMap = mockSiteMapTree;
      component.renderPathInfo = mockRenderPathInfo;
      component.ngOnChanges({
        renderPathInfo: {  } as SimpleChange,
        siteMap: {  } as SimpleChange,
      });

      fixture.detectChanges();
    });

    it('should show the tree when loaded', () => {
      expect(componentEl.querySelector('mat-tree')).toMatchSnapshot();
    });

    it('should indicate selected node', () => {
      component.renderPathInfo = '/idTest4';
      component.ngOnChanges({
        renderPathInfo: {  } as SimpleChange,
      });

      fixture.detectChanges();

      expect(componentEl.querySelector('mat-tree')).toMatchSnapshot();
    });

    it('should load page on node click', () => {
      const rootNode = componentEl.querySelector<HTMLElement>('.qa-tree-node-name-root');

      rootNode?.click();

      expect(iframeService.load).toHaveBeenCalledWith(mockSiteMapTree[0].renderPathInfo);
    });
  });

  describe('search', () => {
    beforeEach(() => {
      component.siteMap = mockSiteMapTree;
      component.renderPathInfo = mockRenderPathInfo;
      component.ngOnChanges({
        renderPathInfo: {} as SimpleChange,
        siteMap: {} as SimpleChange,
      });

      fixture.detectChanges();
    });

    it('should hide not matching nodes', fakeAsync(() => {
      component.search$.next('Test4');
      tick(500);
      fixture.detectChanges();

      expect(componentEl.querySelector<HTMLElement>('.qa-tree-node-name-nameTest2')?.className).toContain('hidden');
      expect(componentEl.querySelector<HTMLElement>('.qa-tree-node-name-nameTest4')?.className).not.toContain('hidden');
    }));

    it('should be case insensitive', fakeAsync(() => {
      component.search$.next('nametest2');
      tick(500);
      fixture.detectChanges();

      expect(componentEl.querySelector<HTMLElement>('.qa-tree-node-name-nameTest2')?.className).not.toContain('hidden');
    }));

    it('should keep parent nodes', fakeAsync(() => {
      component.search$.next('Test3');
      tick(500);
      fixture.detectChanges();

      const parentNode = componentEl.querySelector<HTMLElement>('.qa-tree-node-name-nameTest2');

      expect(parentNode?.className).not.toContain('hidden');
      expect(parentNode?.className).toContain('disabled');
    }));

    it('should keep descendant nodes', fakeAsync(() => {
      component.search$.next('Test2');
      tick(500);
      fixture.detectChanges();

      expect(componentEl.querySelector<HTMLElement>('.qa-tree-node-name-nameTest3')?.className).not.toContain('hidden');
    }));
  });
});
