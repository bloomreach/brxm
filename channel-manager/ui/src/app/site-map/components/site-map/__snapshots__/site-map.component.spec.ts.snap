// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteMapComponent no tree items should display empty tree 1`] = `
<mat-tree
  class="mat-tree"
  role="tree"
>
  
</mat-tree>
`;

exports[`SiteMapComponent tree behavior should indicate selected node 1`] = `
<mat-tree
  class="mat-tree"
  role="tree"
>
  
  <mat-tree-node
    aria-expanded="true"
    class="mat-tree-node qa-tree-node-name-root hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="group"
    style="padding-left: 0px;"
  >
    
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate expanded mat-icon-no-color"
      mattreenodetoggle=""
      role="img"
      svgicon="chevron-right"
    >
      <svg
        class="fake-testing-svg"
        fit=""
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        width="100%"
      />
    </mat-icon>
     root 
  </mat-tree-node>
  <mat-tree-node
    aria-expanded="true"
    class="mat-tree-node qa-tree-node-name-nameTest2 hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="group"
    style="padding-left: 24px;"
  >
    
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate expanded mat-icon-no-color"
      mattreenodetoggle=""
      role="img"
      svgicon="chevron-right"
    >
      <svg
        class="fake-testing-svg"
        fit=""
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        width="100%"
      />
    </mat-icon>
     nameTest2 
  </mat-tree-node>
  <mat-tree-node
    aria-expanded="true"
    aria-level="2"
    class="mat-tree-node qa-tree-node-name-nameTest3 hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="treeitem"
    style="padding-left: 48px;"
  >
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate material-icons mat-icon-no-color"
      role="img"
    />
    
     nameTest3 
  </mat-tree-node>
  <mat-tree-node
    aria-expanded="true"
    aria-level="1"
    class="mat-tree-node qa-tree-node-name-nameTest4 hidden disabled selected"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="treeitem"
    style="padding-left: 24px;"
  >
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate material-icons mat-icon-no-color"
      role="img"
    />
    
     nameTest4 
  </mat-tree-node>
</mat-tree>
`;

exports[`SiteMapComponent tree behavior should show the tree when loaded 1`] = `
<mat-tree
  class="mat-tree"
  role="tree"
>
  
  <mat-tree-node
    aria-expanded="true"
    class="mat-tree-node qa-tree-node-name-root hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="group"
    style="padding-left: 0px;"
  >
    
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate expanded mat-icon-no-color"
      mattreenodetoggle=""
      role="img"
      svgicon="chevron-right"
    >
      <svg
        class="fake-testing-svg"
        fit=""
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        width="100%"
      />
    </mat-icon>
     root 
  </mat-tree-node>
  <mat-tree-node
    aria-expanded="true"
    class="mat-tree-node qa-tree-node-name-nameTest2 hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="group"
    style="padding-left: 24px;"
  >
    
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate expanded mat-icon-no-color"
      mattreenodetoggle=""
      role="img"
      svgicon="chevron-right"
    >
      <svg
        class="fake-testing-svg"
        fit=""
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        width="100%"
      />
    </mat-icon>
     nameTest2 
  </mat-tree-node>
  <mat-tree-node
    aria-expanded="true"
    aria-level="2"
    class="mat-tree-node qa-tree-node-name-nameTest3 selected hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="treeitem"
    style="padding-left: 48px;"
  >
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate material-icons mat-icon-no-color"
      role="img"
    />
    
     nameTest3 
  </mat-tree-node>
  <mat-tree-node
    aria-expanded="false"
    aria-level="1"
    class="mat-tree-node qa-tree-node-name-nameTest4 hidden disabled"
    mattreenodepadding=""
    mattreenodepaddingindent="24"
    role="treeitem"
    style="padding-left: 24px;"
  >
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate material-icons mat-icon-no-color"
      role="img"
    />
    
     nameTest4 
  </mat-tree-node>
</mat-tree>
`;
